packages/patrol/lib/src/custom_finders/patrol_finder.dart

```diff
@@ -474,15 +478,33 @@ extension ActionCombiner on Future<PatrolFinder> {
   /// visible.
   Future<void> enterText(
     String text, {
-    bool? andSettle,
+    @Deprecated('Use settleBeahvior argument instead') bool? andSettle,
+    SettlePolicy? settlePolicy,
     Duration? visibleTimeout,
     Duration? settleTimoeut,
   }) async {
     await (await this).enterText(
       text,
-      andSettle: andSettle,
+      settlePolicy: _chooseSettlePolicy(andSettle, settlePolicy),
       visibleTimeout: visibleTimeout,
       settleTimeout: settleTimoeut,
     );
   }
 }
+
+SettlePolicy? _chooseSettlePolicy(
+  bool? andSettle,
+  SettlePolicy? settlePolicy,
+) {
+  SettlePolicy? settle;
+  if (andSettle == null) {
+    settle = settlePolicy;
+  } else {
+    if (andSettle) {
+      settle = SettlePolicy.settle;
+    } else {
+      settle = SettlePolicy.noSettle;
+    }
+  }
+  return settle;
+}
```

Comment:
Instead of duplicating the code in `patrol_finder.dart` and `patrol_tester.dart` do this:
- move it to a common location
- make it public (remove the leading `_`)
- mark it `@internal`

Reply:
done